### ISSUE TITLE
Task vNext - Update WMR to use platform code for runtime

### ISSUE THREAD

**Issue by SimonDarksideJ** [Microsoft/MixedRealityToolkit-Unity#2248]
## Overview
The Windows Mixed Reality Device / Controllers currently use the InteractionManager provided by Unity.  This needs to be updated to enable use of the Windows.Spatial namespace in the player layer

## Requires
* Updating the InteractionManager events and handlers to use the Windows.Spatial equivalents 
* Building Controller assemblies that will rebuild and automatically wire up their references with the core MRTK assembly.
* Evaluating if any additional features are needed to support (or removal)
* Investigate possible refactoring options for Devices / Controllers

## Acceptance Criteria
- [ ] I can start Unity in UWP mode whilst configured to use the Windows Mixed Reality platform
- [ ] I can view a scene in editor using an Windows Mixed Reality headset
- [ ] I can visualize a default object as a controller
- [ ] Input System actions are generated from the Windows Mixed Reality controller and sent to the Input System
- [ ] All inputs from the selected controller are mapped to the Input System

**Comment by davidkline-ms** [comment]
We should be building on Unity, not replacing it. Right?

**Comment by SimonDarksideJ** [comment]
This is to use the Windows UWP Platform code (as demonstrated in the last shiproom with the "Player" projects) over Unity's implementation.  Giving us greater freedom (and likely performance) over what is available today

**Comment by StephenHodgson** [comment]
I agree, we should be using the native platform code, so we don't have another layer to go through.

When we're done, we will be able to use the native platform code in the editor runtime as well.

**Comment by davidkline-ms** [comment]
Does this imply custom platform code for each device? Not having to write platform code for input, etc is one of the benefits of Unity.

I worry that we are duplicating effort that can best be used elsewhere in the project.

**Comment by StephenHodgson** [comment]
It will not, because of the way we've setup the Input Actions to be device agnostic.

**Comment by StephenHodgson** [comment]
> I worry that we are duplicating effort that can best be used elsewhere in the project.

That was my same thoughts, but Simon is paranoid of Unity, lol.

**Comment by SimonDarksideJ** [comment]
This isn't implemented yet.
The aim is to replace the Unity "Interactionmanager" use with the Windows.Spatial namespace in the player project.

**Comment by davidkline-ms** [comment]
Thanks for clarifying, @SimonDarksideJ
